models/issue_label.go

```diff
@@ -204,12 +237,92 @@ func NewLabels(labels ...*Label) error {
 	return sess.Commit()
 }
 
+// UpdateLabel updates label information.
+func UpdateLabel(l *Label) error {
+	if !LabelColorPattern.MatchString(l.Color) {
+		return fmt.Errorf("bad color code: %s", l.Color)
+	}
+	return updateLabel(x, l)
+}
+
+// DeleteLabel delete a label
+func DeleteLabel(id, labelID int64) error {
+
+	label, err := GetLabelByID(labelID)
+	if err != nil {
+		if IsErrLabelNotExist(err) {
+			return nil
+		}
+		return err
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err = sess.Begin(); err != nil {
+		return err
+	}
+
+	if label.BelongsToOrg() {
+		sess.And("org_id = ?", id)
```

Comment:
Checking the repo/org ID doesn't seem necessary. Once we know the label ID, it should be enough, I think. Am I missing something?

Reply:
Some kind of check is needed because labelID is user supplied in some cases while we set the id via context. If we don't check a user can pass a label to an id they don't have permission to delete.

This code always checked the repoID this change just switches between repo and org.

Reply:
Mmm..... doesn't seem anything like secure... 🤔 For example, comments will be wiped regardless of the provided org/repo ids. IMHO, it's the router's job to decide whether the user can perform the action, and it should call models only when the action is granted.

If the router is at least validating the user permissions on the repo/org, then this function should check the repo/org ID from the label and fail explicitly if it doesn't match with the provided value. But someone _must_ check the user's ability to perform the action.

Reply:
The router does check the ability to perform the action, but all the API asks for is a repo name and label ID. If the user has write permissions for the repo/org settings, then the API check passes and it sends the labelID and repo/org ID along to this function.

Once it gets here, the repo/org id provided is included in the SQL query so a user can't (accidentally or on purpose) delete somebody eases label by providing the wrong labelID. The query guarantees that a user can only delete a label they own because it requires the label to have a repo/org id that they own.

So I guess I would consider that checking if the id matches the label id provided and failing immediately as you've described, since thats what happens. 

I don't think its possible to delete a comment regardless either because as far as I read it if the query doesn't find a label to delete it will return an err before getting to that part. Maybe I'm misunderstanding that xorm code though.

Anyways If the id check was omitted as originally suggested, then it would be insecure and possible to delete anothers label.

We could check the ID before the query instead of within the query if that would be preferable and just error out at that point?

Reply:
I just added the check before the query instead since it seems preferable to not even try and find a label if we know the information is wrong.

Reply:
`Delete()` doesn't fail if no records match the criteria. It just deletes nothing, so if the org/repo ID is wrong, no label is deleted, but since comments are deleted by label ID, comments _will_ be deleted even if the org/repo ID doesn't match the label.

Reply:
ah OK, good to know! I've changed it already so that won't happen any more then and now it is a bugfix too.